snuba_sdk/timeseries.py

```diff
@@ -84,7 +84,7 @@ class Timeseries:
     """
 
     metric: Metric
-    aggregate: str
+    aggregate: str | None = None
```

Comment:
I don't think this is a good idea. I actually disagree with the strategy you're taking here: I think `p90(transaction.duration)` should be `Formula(name="p90", parameters=tuple("transaction.duration")`. The reason is that I think a Timeseries should always have an aggregate.

Reply:
Sure, the problem with that is we lose support for filtering and grouping by inside aggregate. 
For example: `p90(transaction.duration{foo:bar})`

In the suggested strategy, this would become `Formula(name="p90", parameters=tuple("transaction.duration{foo:bar}")`. My thought is that we resolve into a `Timeseries` (with an empty `aggregate` for now) we could plug the `quantiles` aggregate later in the metrics layer.